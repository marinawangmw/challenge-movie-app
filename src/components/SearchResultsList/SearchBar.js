import React from 'react';
import { StyleSheet, View, Image, TextInput } from 'react-native';
import { useTheme } from '@react-navigation/native';
import { typography, spacing } from '@/theme';
import { searchIcon, audioIcon } from '@/assets';
import { en } from '@/localization/en';

export const SearchBar = ({ input, setInput }) => {
  const { colors } = useTheme();

  return (
    <View style={[styles.searchBar, { backgroundColor: colors.card }]}>
      <Image source={searchIcon} style={styles.icon} accessibilityIgnoresInvertColors />
      <TextInput
        autoFocus
        placeholder={en.search.inputPlaceholder}
        placeholderTextColor={colors.text}
        value={input}
        onChangeText={setInput}
        style={[styles.input, { color: colors.text }, typography.text]}
      />
      <Image source={audioIcon} style={styles.icon} accessibilityIgnoresInvertColors />
    </View>
  );
};

const styles = StyleSheet.create({
  searchBar: {
    width: '100%',
    paddingVertical: spacing.m,
    paddingHorizontal: spacing.xs,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  input: {
    flex: 1,
  },
  icon: {
    height: 15,
    width: 15,
    marginHorizontal: spacing.s,
    resizeMode: 'contain',
  },
  title: {
    paddingVertical: spacing.m,
    paddingHorizontal: spacing.s,
  },
});
