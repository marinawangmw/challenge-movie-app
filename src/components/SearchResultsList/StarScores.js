import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { starIcon, halfStarIcon } from '@/assets';

const ScoreTenBase = 10;
const ScoreFiveBase = 5;
const HalfStarMinimum = 5;

export const StarScores = ({ voteAverage = 0 }) => {
  const score = (voteAverage / ScoreTenBase) * ScoreFiveBase;
  const scoreInt = Math.trunc(score);
  const scoreDecimals = score - scoreInt;
  const scoreDecimalFirstDigit = scoreDecimals.toString().slice(2, 3);

  const starPlaceholder = Array(scoreInt).fill(1);

  if (scoreDecimalFirstDigit >= HalfStarMinimum) {
    starPlaceholder.push(0);
  }

  const starImage = (idx, icon) => (
    <Image
      key={idx}
      source={icon}
      accessibilityIgnoresInvertColors
      resizeMode="cover"
      style={styles.starIcon}
    />
  );

  const renderStars = () => {
    return starPlaceholder.map((star, idx) => {
      if (star) {
        return starImage(idx, starIcon);
      }
      return starImage(idx, halfStarIcon);
    });
  };

  return <View style={styles.container}>{renderStars()}</View>;
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  starIcon: {
    height: 15,
    width: 15,
    marginRight: 5,
  },
});
