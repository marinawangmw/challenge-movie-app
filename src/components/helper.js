import { Linking } from 'react-native';
import { Config } from 'react-native-config';
import Toast from 'react-native-toast-message';
import { getMovieTrailer } from '@/controllers/MovieController';

export const playTrailer = async (id) => {
  try {
    const trailerData = await getMovieTrailer(id);

    if (trailerData.results.length) {
      const url = Config.VIDEO_EXTERNAL_BASE_URL + trailerData.results[0].key;

      const supported = await Linking.canOpenURL(url);

      if (supported) {
        await Linking.openURL(url);
      } else {
        Toast.show({
          type: 'error',
          text1: "Something's wrong",
          text2: `Don't know how to open this URL: ${url}`,
        });
      }
    } else {
      Toast.show({
        type: 'error',
        text1: 'Oops',
        text2: "There's no trailer found for this movie",
      });
    }
  } catch (e) {
    console.log(e);
  }
};
