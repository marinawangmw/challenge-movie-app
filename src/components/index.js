export { Button } from '@/components/Button';
export { ErrorView } from '@/components/ErrorView';
export { TabBarIcon } from '@/components/TabBarIcon';
export { TextField } from '@/components/TextField';
export { Hero } from '@/components/Hero';
export { Header } from '@/components/Header';
export { MovieList } from '@/components/MovieList';
