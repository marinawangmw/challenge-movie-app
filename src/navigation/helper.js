export const screenOptions = (textColor) => ({
  headerShown: true,
  headerBackTitle: '',
  headerTintColor: textColor,
});
