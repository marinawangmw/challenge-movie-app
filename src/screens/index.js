export { Home } from '@/screens/Home/Home';
export { Login } from '@/screens/Login/Login';
export { Profile } from '@/screens/Profile/Profile';
export { Loading } from '@/screens/Loading/Loading';
export { MovieCollection } from '@/screens/MovieCollection/MovieCollection';
export { MovieDetail } from '@/screens/MovieDetail/MovieDetail';
