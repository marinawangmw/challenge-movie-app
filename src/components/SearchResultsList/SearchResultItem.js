import React from 'react';
import { TouchableOpacity, StyleSheet, View, Image, Text } from 'react-native';
import { useTheme } from '@react-navigation/native';
import { Config } from 'react-native-config';
import { StarScores } from './StarScores';
import { playIcon } from '@/assets';
import { NAVIGATION } from '@/constants';
import { spacing, typography } from '@/theme';

const SearchResultItem = ({ item, navigation }) => {
  const { colors } = useTheme();

  const handleNavigate = (itemData) => {
    navigation.navigate(NAVIGATION.movieDetail, {
      movieDetails: itemData,
    });
  };

  return (
    <TouchableOpacity
      onPress={() => handleNavigate(item.item)}
      style={[styles.resultBlock, { backgroundColor: colors.card }]}
    >
      <Image
        source={{ uri: Config.IMAGE_API_BASE_URL + item.item.poster_path }}
        accessibilityIgnoresInvertColors
        resizeMode="cover"
        style={styles.resultPoster}
      />

      <View style={styles.resultDetail}>
        <View style={styles.resultDetailUp}>
          <Text style={[typography.text, { color: colors.text }, styles.itemTitle]}>
            {item.item.title}
          </Text>
        </View>

        <View style={styles.resultDetailBottom}>
          <StarScores voteAverage={item.item.vote_average} />
          <Image
            source={playIcon}
            style={styles.playIcon}
            accessibilityIgnoresInvertColors
            resizeMode="contain"
          />
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default SearchResultItem;

const styles = StyleSheet.create({
  resultBlock: {
    margin: 2,
    flexDirection: 'row',
  },
  resultPoster: {
    width: 160,
    height: 95,
  },
  resultDetail: {
    flexDirection: 'column',
    padding: spacing.s,
    flex: 1,
    flexWrap: 'wrap',
  },
  resultDetailUp: {
    paddingBottom: spacing.s,
    width: '100%',
  },
  itemTitle: {
    flexWrap: 'wrap',
  },
  resultDetailBottom: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  playIcon: {
    width: 20,
    height: 20,
  },
});
