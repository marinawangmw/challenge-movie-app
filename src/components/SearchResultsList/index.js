import React from 'react';
import { StyleSheet, Text, FlatList } from 'react-native';
import { useSelector } from 'react-redux';
import { useTheme } from '@react-navigation/native';
import SearchResultItem from './SearchResultItem';
import { Loading } from '@/screens';
import { MessageBanner } from '@/components';
import { typography, spacing } from '@/theme';
import { en } from '@/localization/en';
import { TYPES } from '@/actions/MovieListActions';
import { isLoadingSelector } from '@/selectors/StatusSelectors';

export const SearchResultsList = ({ data, navigation, title }) => {
  const { colors } = useTheme();
  const isLoading = useSelector((state) => isLoadingSelector([TYPES.FETCH_MOVIE_LISTS], state));

  const renderEmptyMessage = () => {
    return <MessageBanner message={en.search.noObjectMessage} customStyles={message} />;
  };

  const renderHeader = () => {
    if (isLoading) {
      return <Loading />;
    }

    if (title) {
      return <Text style={[{ color: colors.text }, styles.title, typography.title]}>{title}</Text>;
    }

    return null;
  };

  return (
    <FlatList
      data={data}
      renderItem={(item) => <SearchResultItem navigation={navigation} item={item} />}
      keyExtractor={(_item, idx) => idx}
      contentContainerStyle={styles.container}
      ListHeaderComponent={renderHeader}
      ListEmptyComponent={renderEmptyMessage}
    />
  );
};

export const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    padding: spacing.s,
  },
  title: {
    paddingVertical: spacing.s,
  },
});

export const message = StyleSheet.create({
  messageContainer: {
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
