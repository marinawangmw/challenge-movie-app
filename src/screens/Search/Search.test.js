import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Search } from '@/screens';
import { SearchResultsList } from '@/components';
import { SearchBar } from '@/components/SearchResultsList/SearchBar';
import SearchResultItem from '@/components/SearchResultsList/SearchResultItem';
import { StarScores } from '@/components/SearchResultsList/StarScores';
import { withProviders } from '@/test-utils';
import { navigation, movieList } from '@/test-utils/helper';

const fakeEmptyStore = {
  error: {},
  status: {},
  user: {},
  movieList: {
    errorMessage: undefined,
    myList: [],
    heroPoster: { posterMovie: null, posterGenres: [] },
    movieLists: [],
    similarMovies: {
      page: 1,
      results: [],
    },
    searchResults: [],
  },
};

const fakeStoreWithData = {
  error: {},
  status: {},
  user: {},
  movieList: {
    errorMessage: undefined,
    myList: [],
    heroPoster: { posterMovie: movieList.movieList[0], posterGenres: ['drama', 'romance'] },
    movieLists: [],
    similarMovies: {
      page: 1,
      results: [],
    },
    searchResults: [],
  },
};

describe('Search component', () => {
  it('renders correcly with empty hero poster', () => {
    const { toJSON } = render(
      withProviders(<Search navigation={navigation} />, {
        initialState: fakeEmptyStore,
      })
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correcly with data on hero poster', () => {
    const { toJSON } = render(
      withProviders(<Search navigation={navigation} />, {
        initialState: fakeStoreWithData,
      })
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Search bar component', () => {
  it('renders correctly and match snapshot', () => {
    const input = '';
    const setInput = () => null;

    const { toJSON } = render(<SearchBar input={input} setInput={setInput} />);
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Search results list', () => {
  it('renders correctly and match snapshot', () => {
    const { toJSON } = render(
      withProviders(<SearchResultsList navigation={navigation} />, {
        initialState: fakeEmptyStore,
      })
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with data', () => {
    const data = movieList.movieList;
    const { toJSON } = render(
      withProviders(<SearchResultsList data={data} navigation={navigation} />, {
        initialState: fakeStoreWithData,
      })
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with title', () => {
    const title = 'Popular results';
    const { queryByText } = render(
      withProviders(<SearchResultsList navigation={navigation} title={title} />, {
        initialState: fakeEmptyStore,
      })
    );
    expect(queryByText('Popular results')).not.toBeNull();
  });
});

describe('Search results item', () => {
  it('renders correctly and match snapshot', () => {
    const item = { item: movieList.movieList[0] };
    const { toJSON } = render(
      withProviders(<SearchResultItem item={item} navigation={navigation} />, {
        initialState: fakeEmptyStore,
      })
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('Star scores component', () => {
  it('renders correctly with no prop passed', () => {
    const tree = renderer.create(<StarScores />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly passing a prop', () => {
    const voteAverage = 9.9;

    const tree = renderer.create(<StarScores voteAverage={voteAverage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const voteAverage = 9.9;

    const component = shallow(<StarScores voteAverage={voteAverage} />);

    expect(component.length).toBe(1);
  });

  it('renders correct amount of stars', () => {
    const voteAverage = 9.9;

    const component = shallow(<StarScores voteAverage={voteAverage} />);

    expect(component.find({ testID: 'stars' }).length).toBe(5);
  });
});
