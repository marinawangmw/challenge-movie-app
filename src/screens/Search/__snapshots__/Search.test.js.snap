// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component renders correctly and match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
        "paddingVertical": 24,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
      },
    ]
  }
>
  <Image
    accessibilityIgnoresInvertColors={true}
    source={
      Object {
        "testUri": "../../../src/assets/search.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginHorizontal": 16,
        "resizeMode": "contain",
        "width": 15,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoFocus={true}
    onChangeText={[Function]}
    placeholder="Search for a movie that you love..."
    placeholderTextColor="rgb(28, 28, 30)"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
        Object {
          "fontSize": 16,
          "fontWeight": "400",
        },
      ]
    }
    testID="searchInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <Image
    accessibilityIgnoresInvertColors={true}
    source={
      Object {
        "testUri": "../../../src/assets/mic.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginHorizontal": 16,
        "resizeMode": "contain",
        "width": 15,
      }
    }
  />
</View>
`;

exports[`Search component renders correcly with data on hero poster 1`] = `null`;

exports[`Search component renders correcly with empty hero poster 1`] = `null`;

exports[`Search results item renders correctly and match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "flexDirection": "row",
      "margin": 2,
      "opacity": 1,
    }
  }
>
  <Image
    accessibilityIgnoresInvertColors={true}
    resizeMode="cover"
    source={
      Object {
        "uri": "undefined/aJbeDPUwDe9MLpseTFrNTBaQPzf.jpg",
      }
    }
    style={
      Object {
        "height": 95,
        "width": 160,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "wrap",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "paddingBottom": 16,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
            },
            Object {
              "color": "rgb(28, 28, 30)",
            },
            Object {
              "flexWrap": "wrap",
            },
          ]
        }
      >
        The Woman in Black
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/star.png",
            }
          }
          style={
            Object {
              "height": 15,
              "marginRight": 5,
              "width": 15,
            }
          }
          testID="stars"
        />
        <Image
          accessibilityIgnoresInvertColors={true}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/star.png",
            }
          }
          style={
            Object {
              "height": 15,
              "marginRight": 5,
              "width": 15,
            }
          }
          testID="stars"
        />
        <Image
          accessibilityIgnoresInvertColors={true}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/star.png",
            }
          }
          style={
            Object {
              "height": 15,
              "marginRight": 5,
              "width": 15,
            }
          }
          testID="stars"
        />
      </View>
      <Image
        accessibilityIgnoresInvertColors={true}
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/play.png",
          }
        }
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Search results list renders correctly and match snapshot 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  ListHeaderComponent={[Function]}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "padding": 16,
    }
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
          }
        }
      >
        Nothing found
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Search results list renders correctly with data 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  ListHeaderComponent={[Function]}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "padding": 16,
    }
  }
  data={
    Array [
      Object {
        "adult": false,
        "backdrop_path": "/ziTxBaF1AhbUwcv2IBiFJAf54qD.jpg",
        "id": 65086,
        "overview": "The story follows a young lawyer, Arthur Kipps.",
        "popularity": 16.334,
        "poster_path": "/aJbeDPUwDe9MLpseTFrNTBaQPzf.jpg",
        "release_date": "2012-02-03",
        "title": "The Woman in Black",
        "vote_average": 6.11,
      },
      Object {
        "adult": false,
        "id": 102899,
        "overview": "Armed with the astonishing ability to shrink in scale but increase in strength.",
        "popularity": 69.704,
        "poster_path": "/rS97hUJ1otKTTripGwQ0ujbuIri.jpg",
        "release_date": "2015-07-14",
        "title": "Ant-Man",
        "vote_average": 7.088,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "flexDirection": "row",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          resizeMode="cover"
          source={
            Object {
              "uri": "undefined/aJbeDPUwDe9MLpseTFrNTBaQPzf.jpg",
            }
          }
          style={
            Object {
              "height": 95,
              "width": 160,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "wrap",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "paddingBottom": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                  },
                  Object {
                    "flexWrap": "wrap",
                  },
                ]
              }
            >
              The Woman in Black
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/star.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/star.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/star.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
            </View>
            <Image
              accessibilityIgnoresInvertColors={true}
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../src/assets/play.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "flexDirection": "row",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          resizeMode="cover"
          source={
            Object {
              "uri": "undefined/rS97hUJ1otKTTripGwQ0ujbuIri.jpg",
            }
          }
          style={
            Object {
              "height": 95,
              "width": 160,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "flexWrap": "wrap",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "paddingBottom": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                  },
                  Object {
                    "flexWrap": "wrap",
                  },
                ]
              }
            >
              Ant-Man
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/star.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/star.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/star.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
              <Image
                accessibilityIgnoresInvertColors={true}
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/halfStar.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "marginRight": 5,
                    "width": 15,
                  }
                }
                testID="stars"
              />
            </View>
            <Image
              accessibilityIgnoresInvertColors={true}
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../src/assets/play.png",
                }
              }
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Star scores component renders correctly passing a prop 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Image
    accessibilityIgnoresInvertColors={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/star.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginRight": 5,
        "width": 15,
      }
    }
    testID="stars"
  />
  <Image
    accessibilityIgnoresInvertColors={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/star.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginRight": 5,
        "width": 15,
      }
    }
    testID="stars"
  />
  <Image
    accessibilityIgnoresInvertColors={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/star.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginRight": 5,
        "width": 15,
      }
    }
    testID="stars"
  />
  <Image
    accessibilityIgnoresInvertColors={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/star.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginRight": 5,
        "width": 15,
      }
    }
    testID="stars"
  />
  <Image
    accessibilityIgnoresInvertColors={true}
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/assets/halfStar.png",
      }
    }
    style={
      Object {
        "height": 15,
        "marginRight": 5,
        "width": 15,
      }
    }
    testID="stars"
  />
</View>
`;

exports[`Star scores component renders correctly with no prop passed 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
/>
`;
